### DISCUSSION TITLE
eng-rus localizatios

### DISCUSSION THREAD

**Discussion by CheKecha19** [ocornut/imgui#4037] [Old/Archived discussions]
I have a question for you: can I use Russian in your ImGui development and how can I do it?
The GUI breaks when trying to change some function values from English to Russian (keyboard key values). I tried to change the encoding inside the source code, but it didn't give me any results.
the SETLOCALE function also did not give the expected result, although it is intended for such localization.

![image](https://user-images.githubusercontent.com/61182912/114358968-2f7e7a00-9b7c-11eb-83ec-493643d91df0.png)
![image](https://user-images.githubusercontent.com/61182912/114360065-4a9db980-9b7d-11eb-8880-bc655e3ee302.png)

this pops up during compilation after changing Latin characters to Cyrillic

the Internet says that utf-8, which you use, supports Cyrillic, but, in fact, it turns out that it does not

I hope you can help with some advice

**Comment by rokups** [comment]
It is not clear what you are trying to do here. If you want to render russian text - see [fonts documentation](https://github.com/ocornut/imgui/blob/master/docs/FONTS.md). If you are trying to convert source code to russian - no compiler would accept that because it is against C++ spec.

**Comment by iocafe** [comment]
Hi,
I hope this helps. ImGui works fine for Russian, I believe it works for at least all languages which can be written from left to right. Not so sure about Arabic and Hebrew. 

Google's NOTO fonts can be downloaded from https://www.google.com/get/noto/help/install/ as one big zip file.
Here only one of the files within zip is needed, I picked up NotoSansDisplay-Regular.ttf.

Use something like io.Fonts->AddFontFromFileTTF("/coderoot/bluetree/egui/fonts/NotoSansDisplay-Regular.ttf", 20.0f, NULL, io.Fonts->GetGlyphRangesCyrillic()); to load the Cyrillic glyphs to ImGui (latin ones are loaded by default).

If Russian string literals are needed within C/C++ code, portable way is to code these as escape sequences:
const char myrussiantext[] = "\xd0\xbc\xd0\xb0\xd1\x82\xd0\xb5\xd1\x80\xd0\xb8\xd0\xbd\xd1\x81\xd0\xba\xd0\xb0\xd1\x8f\x20\xd0\xbd\xd0\xbe\xd1\x87\xd1\x8c";

Tool to convert Unicode text (Chinese, Russian, etc.) to C escape sequences: https://onlineunicodetools.com/escape-unicode
Select "use bytes", UTF8 and no separator. The escaped unicode can be copy/pasted into code. 

![image](https://user-images.githubusercontent.com/54875976/115121941-4dd0e500-9f83-11eb-8ea9-11b6652ca971.png)
